### ISSUE TITLE
Assign public IPs to EKS workers in private subnets.

### ISSUE THREAD

**Issue by ozbillwang** [terraform-aws-modules/terraform-aws-eks#58]
# I have issues

I created EKS cluster in private subnets, we also have discussed about this topic in several tickets, we agree to create EKS workers in private subnets only.

Now it's time to decide, should we keep the feature to assign public IP to EKS workers?

If it is not required any more, I will raise PR to remove this line directly. Otherwise, I have to update with a condition, which way you like?

```diff
resource "aws_launch_configuration" "workers" {
  name_prefix                 = "${var.cluster_name}-${lookup(var.worker_groups[count.index], "name", count.index)}"
-  associate_public_ip_address = "${lookup(var.worker_groups[count.index], "public_ip", lookup(var.workers_group_defaults, "public_ip"))}"
  security_groups             = ["${local.worker_security_group_id}"]
  iam_instance_profile        = "${aws_iam_instance_profile.workers.id}"
  image_id                    = "${lookup(var.worker_groups[count.index], "ami_id", data.aws_ami.eks_worker.id)}"
  instance_type               = "${lookup(var.worker_groups[count.index], "instance_type", lookup(var.workers_group_defaults, "instance_type"))}"
```
## I'm submitting a...

* [X] bug report
* [X] feature request

## What is the current behavior?

When create workers in private subnets, public IPs are assigned to these workers.

## Are you able to fix this problem and submit a PR? Link here if you have already.

Yes, I will
## Environment details

* Affected module version: v1.1.0
* OS: ubuntu
* Terraform version: 0.11.7

**Comment by brandoconnor** [comment]
Sorry for the delay here. Yep I also agree this should be configurable. Feel free to submit a PR with WIP in the title for this work: https://github.com/ozbillwang/terraform-aws-eks/commit/2173bbcfa926e48785c6ed6cd8dc935269f6a7c9

We're headed the right direction with that. If what you have is as good as terraform allows, that'd still be a fine addition.

**Comment by ozbillwang** [comment]
Seems someone had implemented it with PR #18
